src/scripting/game_lua_kernel.cpp

```diff
@@ -1043,7 +1043,7 @@ int game_lua_kernel::impl_get_terrain_info(lua_State *L)
 {
 	char const *m = luaL_checkstring(L, 2);
 	t_translation::terrain_code t = t_translation::read_terrain_code(m);
-	if (t == t_translation::NONE_TERRAIN) return 0;
+	if (t == t_translation::NONE_TERRAIN || !board().map().tdata()->is_known(t)) return 0;
 	const terrain_type& info = board().map().tdata()->get_terrain_info(t);
```

Comment:
Instead of checking before this line, maybe check afterwards with
```
if (!info.is_nonnull()) {
    return 0;
}
```

Reply:
I find that less clear. That function even has a todo comment to make it clearer.

Reply:
`is_known` is what is called before the game tells you there is an unknown tile in the map. So to me that is the canonical check.